src/library_wasi.js

```diff
@@ -15,16 +15,18 @@ var WasiLibrary = {
 #endif
   },
 
-  $getEnvStrings__deps: ['$ENV', '$getExecutableName'],
+  $getEnvStrings__deps: ['$ENV', '$getExecutableName', '$emscriptenGetTimeZone'],
   $getEnvStrings: function() {
     if (!getEnvStrings.strings) {
       // Default values.
 #if !DETERMINISTIC
       // Browser language detection #8751
       var lang = ((typeof navigator == 'object' && navigator.languages && navigator.languages[0]) || 'C').replace('-', '_') + '.UTF-8';
+      var tz = emscriptenGetTimeZone();
```

Comment:
Would it make more sense to put the DETERMINISTIC check inside `emscriptenGetTimeZone` ? Just thinking later when applying the `library_fs` changes?